### PR TITLE
Blending fix + ignore irradiance per material

### PR DESCRIPTION
**Pull request by Naxela**
I thought I had made a pull request about this earlier, but turns out I had forgotten to.

In any case, it's a fix for https://github.com/armory3d/armory/issues/1615 which involves issues with blending. On deferred blending causes a setPipeline error, whilst forward displays nothing.

Before on forward:

Forward

After:

Fixed

### CODE REVIEW

**Comment by Naxela** [comment]
Additional commit:

- Add option to ignore irradiance per material

For baked environments, the option to ignore irradiance is nice to have, since when SH irradiance is applied to baked objects the static environments end up looking a bit too saturated and contrasted with wrong lighting:

image

I'm unsure of whether the implementation is good though, since it's using the available space in gbuffer2 (if I understood it correctly, object motion blur uses the rg values, the unused b value is then used for designated which object to be ignored by the environment lighting). I'm not sure if there's some overhead when using the existing buffer or if it would be better to create a separate rendertarget for this, since it's not available for forward rendering either (similar to object motion blur).

With static environments + irradiance applied - Notice the wrong darkening on the walls:

image

and without it looks more natural. This is more apparent in high contrast scenes:

image

In short, the point was to differentiate between static environments with lightmaps and mobile/dynamic objects inside the environment which has the SH coefficients from the interior rendered out and applied on runtime.

**Comment by luboslenco** [comment]
Looks great, thanks!
